### ISSUE TITLE
Async Clipboard API is "all red"

### ISSUE THREAD

**Issue by captainbrosset** [web-platform-dx/web-features#1249]
Reported by @tomayac here: https://toot.cafe/@tomayac/112614061442371457

See the status: https://web-platform-dx.github.io/web-features-explorer/features/async-clipboard/

Compare to Can I Use: https://caniuse.com/async-clipboard

Here are the BCD keys used to compute our status:

- api.Clipboard
- api.Clipboard.read
- api.Clipboard.readText
- api.Clipboard.write
- api.Clipboard.writeText
- api.ClipboardEvent
- api.ClipboardEvent.ClipboardEvent
- api.ClipboardEvent.clipboardData
- api.ClipboardItem
- api.ClipboardItem.ClipboardItem
- api.ClipboardItem.getType
- api.ClipboardItem.presentationStyle
- api.ClipboardItem.types
- api.Navigator.clipboard
- api.Permissions.permission_clipboard-read

**Comment by ddbeck** [comment]
The overall non-support is due to BCD reporting that:

- Firefox and Safari don't support the `clipboard-read` permission (`api.Permissions.permission_clipboard-read`).
- Chrome does not support `ClipboardItem`'s `presentationStyle` property and only partially supports `read` and `readText` methods (`api.Clipboard.read*`).

It's unclear to me whether BCD, caniuse, or both have errors. BCD and caniuse differ significantly. Major differences are:

- `readText`: BCD says this is partially supported in Chrome, caniuse says Firefox doesn't support it.
- `writeText`: BCD says this is fully supported in Safari for iOS, but caniuse says it's unsupported.

Resolving this would be a three step process:

- [ ] Test and (if applicable) correct the data in BCD.

  The following issues are open on BCD for clipboard data:

  - https://github.com/mdn/browser-compat-data/issues/22096
  - https://github.com/mdn/browser-compat-data/issues/18140
  - https://github.com/mdn/browser-compat-data/issues/7991
  
  I'd expect that resolving them would go a long way toward resolving this issue in general.

- [ ] If this feature is still truly partially supported in any of the core browsers, then this is a web-features bug and it's blocked by https://github.com/web-platform-dx/web-features/issues/915. When that's done, mark this as partially supported and explain why. If there are no partials, you can skip this step.

- [ ] Check if there are any remaining discrepancies with caniuse. If so, file an issue (or PR) against to caniuse to resolve those differences.

**Comment by foolip** [comment]
https://github.com/web-platform-dx/web-features/pull/1315 should improve this.

**Comment by foolip** [comment]
Some cleanup of the tags BCD keys in https://github.com/mdn/browser-compat-data/pull/23658.

**Comment by ddbeck** [comment]
Copying from myself on https://github.com/web-platform-dx/web-features/pull/1294#discussion_r1662316952

> Hmm, yeah, that's iffy data. Two thoughts here:
> 
>     * The partial seems to be the requirement to have the (non-standard?) `clipboard-read` permission. I can't find that permission specified anywhere, which makes it hard to understand.
> 
>     * Those media types should be subfeatures of `read` instead of notes. Probably something like `text/plain` and `image/png` ought to be in `compute_from` list?

**Comment by foolip** [comment]
What should we consider the scope of this feature to be?

https://github.com/web-platform-dx/web-features/pull/1315 would make it `navigator.clipboard.writeText()`. At the same time https://caniuse.com/async-clipboard is about both reading and writing, as it treats Safari iOS as a partial implementation with the note "Supports writing to clipboard but not reading."

**Comment by foolip** [comment]
I looked into the Safari iOS issue on caniuse and filed https://github.com/Fyrd/caniuse/issues/7105.

At this point I'm fairly certain that all four methods (`read()`, `readText()`, `write()`, `writeText()`) work in all browsers, but there are differences in permissions, requirements for user interaction, and UX, that developers need to understand in order to write code that works everywhere. We don't have notes in web-features, so BCD is the place for that.

In web-features, since the scope of the feature is all four methods, I think we should set a Baseline low date accordingly. This will have to be done manually because BCD has `partial_implementation` still.

**Comment by foolip** [comment]
I've sent https://github.com/mdn/browser-compat-data/pull/23735 which removes partial impl for Chromium in BCD.

**Comment by ddbeck** [comment]
> I've sent [mdn/browser-compat-data#23735](https://github.com/mdn/browser-compat-data/pull/23735) which removes partial impl for Chromium in BCD.

Superseded by https://github.com/mdn/browser-compat-data/pull/23593.

**Comment by ddbeck** [comment]
Bringing some text from a comment on the YAML file here, so I can use this issue a clearinghouse for all the history of this feature's data issues.

> Differences between web-features and caniuse.com are tracked here:
> https://github.com/Fyrd/caniuse/issues/7105
> https://github.com/Fyrd/caniuse/pull/7106
> https://github.com/Fyrd/caniuse/issues/7107
